samples/13.core-bot/src/main/java/com/microsoft/bot/sample/core/MainDialog.java

```diff
@@ -0,0 +1,183 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.bot.sample.core;
+
+import com.microsoft.bot.builder.MessageFactory;
+import com.microsoft.bot.builder.TurnContext;
+import com.microsoft.bot.dialogs.ComponentDialog;
+import com.microsoft.bot.dialogs.DialogTurnResult;
+import com.microsoft.bot.dialogs.WaterfallDialog;
+import com.microsoft.bot.dialogs.WaterfallStep;
+import com.microsoft.bot.dialogs.WaterfallStepContext;
+import com.microsoft.bot.dialogs.prompts.PromptOptions;
+import com.microsoft.bot.dialogs.prompts.TextPrompt;
+import com.microsoft.bot.schema.Activity;
+import com.microsoft.bot.schema.InputHints;
+import com.microsoft.recognizers.datatypes.timex.expression.TimexProperty;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.apache.commons.lang3.StringUtils;
+
+public class MainDialog extends ComponentDialog {
+    private final FlightBookingRecognizer luisRecognizer;
+
+    public MainDialog(FlightBookingRecognizer withLuisRecognizer, BookingDialog bookingDialog) {
+        super(MainDialog.class.getName());
+
+        luisRecognizer = withLuisRecognizer;
+
+        addDialog(new TextPrompt(TextPrompt.class.getName()));
+        addDialog(bookingDialog);
+        WaterfallStep[] waterfallSteps = {
+            this::introStep,
+            this::actStep,
+            this::finalStep
+        };
+        addDialog(new WaterfallDialog(WaterfallDialog.class.getName(), Arrays.asList(waterfallSteps)));
+
+        // The initial child Dialog to run.
+        setInitialDialogId(WaterfallDialog.class.getName());
+    }
+
+    /**
+     * First step in the waterfall dialog. Prompts the user for a command.
+     * Currently, this expects a booking request, like "book me a flight from Paris to Berlin on march 22"
+     * Note that the sample LUIS model will only recognize Paris, Berlin, New York and London as airport cities.
+     * @param stepContext A {@link WaterfallStepContext}
+     * @return A {@link DialogTurnResult}
+     */
+    private CompletableFuture<DialogTurnResult> introStep(WaterfallStepContext stepContext) {
+        if (!luisRecognizer.isConfigured()) {
+            Activity text = MessageFactory.text("NOTE: LUIS is not configured. To enable all capabilities, add 'LuisAppId', 'LuisAPIKey' and 'LuisAPIHostName' to the appsettings.json file.", null, InputHints.IGNORING_INPUT);
```

Comment:
Checkstyle didn't complain against the lenght of lines like this one?

Reply:
Sure, but this task is only to migrate the `MainDialog`, I didn't check the linter.. this will be accomplished during the fix